### PR TITLE
Add s3 access secret

### PR DESCRIPTION
**Pull request by kuanfandevops**

### CODE REVIEW

File: charts/spilo/values.yaml [hunk 1]
```diff
@@ -48,9 +48,18 @@ continuousArchive:
...
+  # It contains three key/value pairs:
+  #   awsAccessKeyIdName/awsAccessKeyIdVakue
+  #   awsSecretAccessKeyName/awsSecretAccessKeyValue
+  #   awsEndpointName/awsEndpointValue
   s3:
     bucket: ~
     secretName: ~
+    awsAccessKeyIdName: ~
+    awsSecretAccessKeyName: ~
+    awsEndpointName: ~
```
Comment (pbastia) [inline charts/spilo/values.yaml]: We can probably do away with these, and enforce the secret keys to be the same as the environment variables.

```suggestion

```

File: charts/spilo/templates/statefulset.yaml [hunk 2]
```diff
@@ -65,6 +65,21 @@ spec:
         {{- if eq .Values.continuousArchive.storage "s3"}}
         - name: WAL_S3_BUCKET
           value: {{ .Values.walE.s3Bucket | quote }}
+        - name: AWS_ACCESS_KEY_ID
+          valueFrom: 
+            secretKeyRef: 
+              name: {{ .Values.continuousArchive.s3.secretName }}
+              key: {{ .Values.continuousArchive.s3.awsAccessKeyIdName }}
```
Comment (pbastia) [inline charts/spilo/templates/statefulset.yaml]:
```suggestion
              key: AWS_ACCESS_KEY_ID
```
or 
```suggestion
              key: aws_access_key_id
```

**Review by pbastia — CHANGES_REQUESTED** [review]
Nice work!
- I'd suggest enforcing the AWS secret format, this would simplify the chart
- Add a tiny bit of documentation about that in the README
- Your update about making the readiness probe timeout configurable

File: charts/spilo/templates/statefulset.yaml (hunk 2 above)
Comment (kuanfandevops) [reply charts/spilo/templates/statefulset.yaml]: I think this way could be more configurable if the project already has the secret, but it is also true that it is bit confusing.

File: charts/spilo/values.yaml (hunk 1 above)
Comment (kuanfandevops) [reply charts/spilo/values.yaml]: I think this way could be more configurable if the project already has the secret, but it is also true that it is bit confusing. let me think about it.

Comment (kuanfandevops) [reply charts/spilo/values.yaml]: I updated the key names in the secret to be same as the env variables

File: charts/spilo/templates/statefulset.yaml (hunk 2 above)
Comment (kuanfandevops) [reply charts/spilo/templates/statefulset.yaml]: I updated the key names in the secret to be same as the env variables

**Comment by kuanfandevops** [comment]
I added the comments in the values.yaml instead of README file. Please give another review. Also I tested the restore proces s and it worked.

**Review by pbastia — APPROVED** [review]

**Comment by pbastia** [comment]
Looks awesome. Would you mind adding in the `README` the procedure you followed when you successfully restored your DB?

**Comment by kuanfandevops** [comment]
Yes, will do.

**Comment by kuanfandevops** [comment]
@pbastia Applied the usage of parameter useExistingSecret and added documentation of restore. Please have another review.

**Review by pbastia — APPROVED** [review]

**Pull request merged**